### PR TITLE
🌱 : e2e test: do tests with restricted pods

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
do tests with restricted pods

**motivation**
Ensure that the default scaffold works with restricted pods

### CODE REVIEW

**Review by everettraven — CHANGES_REQUESTED** [review]
Overall looks good, just a couple minor nits

File: test/e2e/v3/generate_test.go [hunk 1]
```diff
@@ -228,10 +228,52 @@ Count int `+"`"+`json:"count,omitempty"`+"`"+`
...
+		uncommentPodStandards(kbc)
+	}
+}
+
+func uncommentPodStandards(kbc *utils.TestContext) {
+	configManager := filepath.Join(kbc.Dir, "config", "manager", "manager.yaml")
+	managerAuth := filepath.Join(kbc.Dir, "config", "default", "manager_auth_proxy_patch.yaml")
+
+	//nolint:lll
+	if err := pluginutil.ReplaceInFile(configManager, `# TODO(user): uncomment for common cases that do not require escalating privileges
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply test/e2e/v3/generate_test.go:243]: will not work well here with the TODO

File: test/e2e/v3/plugin_cluster_test.go [hunk 2]
```diff
@@ -160,7 +163,63 @@ var _ = Describe("kubebuilder", func() {
...
+				// Skip if cluster version < 1.19, because securityContext.seccompProfile only works from 1.19
+				// Otherwise, unknown field "seccompProfile" in io.k8s.api.core.v1.PodSecurityContext will be faced
+				// Skip if cluster version >= 1.22 because pre v1 CRDs and webhooks no longer exist.
+				if srvVer := kbc.K8sVersion.ServerVersion; srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() < 19 ||
+					srvVer.GetMajorInt() <= 1 && srvVer.GetMinorInt() >= 22 {
+					Skip(fmt.Sprintf("cluster version %s does not support project defaults "+
+						"and securityContext.seccompProfile", srvVer.GitVersion))
+				}
+
+				GenerateV3(kbc, "v1beta1", true)
```
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply test/e2e/v3/plugin_cluster_test.go:222]: That is with CRD v1beta1

[the remark this answers is not in the document]
Comment (everettraven) [reply test/e2e/v3/plugin_cluster_test.go:222]: It looks like the lines:
https://github.com/kubernetes-sigs/kubebuilder/pull/2732/files/56f0ba5b2730702ed56b5e66d3b5e741ba016c55#diff-5f5cf1a838c4d92e543721d2697ea0e2b0cf57b9f9a3c72bc9fa51a030bc56b8R170-R224 

are repeated tests. I may be wrong but I think all of these tests already exist in these lines:
https://github.com/kubernetes-sigs/kubebuilder/pull/2732/files/56f0ba5b2730702ed56b5e66d3b5e741ba016c55#diff-5f5cf1a838c4d92e543721d2697ea0e2b0cf57b9f9a3c72bc9fa51a030bc56b8R114-R168

File: test/e2e/v3/generate_test.go [hunk 3]
```diff
@@ -228,10 +228,52 @@ Count int `+"`"+`json:"count,omitempty"`+"`"+`
...
+
+	if restrictived {
+		By("uncomment kustomize files to ensure that pods are restricted")
+		uncommentPodStandards(kbc)
+	}
+}
+
+func uncommentPodStandards(kbc *utils.TestContext) {
+	configManager := filepath.Join(kbc.Dir, "config", "manager", "manager.yaml")
+	managerAuth := filepath.Join(kbc.Dir, "config", "default", "manager_auth_proxy_patch.yaml")
```
Comment (rashmigottipati) [inline test/e2e/v3/generate_test.go:240]: Nit:
```suggestion
	managerAuth := filepath.Join(configManager, "default", "manager_auth_proxy_patch.yaml")
```

Comment (camilamacedo86) [reply test/e2e/v3/generate_test.go:240]: We cannot do this change. 
Why?
configManager-> has the full path with the file

File: test/e2e/v3/plugin_cluster_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (camilamacedo86) [reply test/e2e/v3/plugin_cluster_test.go:222]: - the first one test the Operator with v1 CRDs/Webooks and pod restriction
- the second one test the Operator with kustomze v2 changes and pod restriction
- the third one  test the Operator with v1beta1 CRDs/Webooks and pod restriction

In the long run we will need to refractory the tests.  This code is confusing for sure.

**Review by everettraven — APPROVED** [review]
Looks good to me

**Comment by camilamacedo86** [comment]
@rashmigottipati I am moving forward with this one but we can be working on any improvement that you see fit as a fallow up as well.

**Comment by camilamacedo86** [comment]
/hold

to rebase with master

**Review by rashmigottipati — COMMENTED** [review]
/lgtm

**Pull request merged**